src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -612,10 +716,22 @@ public String getContentDisplay()
             {
                 tmp = tmp.replace(emote.getAsMention(), ":" + emote.getName() + ":");
             }
-            for (TextChannel mentionedChannel : getMentionedChannels())
+            for (TextChannel mentionedChannel : getMentionedTextChannels())
+            {
+                tmp = tmp.replace(mentionedChannel.getAsMention(), '#' + mentionedChannel.getName());
+            }
+            for (VoiceChannel mentionedChannel : getMentionedVoiceChannels())
+            {
+                tmp = tmp.replace(mentionedChannel.getAsMention(), mentionedChannel.getName());
```

Comment:
> Also, I'm not sure which characters to use [here](https://github.com/Simon-official/JDA/blob/feature/mentioned-channels/src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java#L723).

```suggestion
                tmp = tmp.replace(mentionedChannel.getAsMention(), '#!' + mentionedChannel.getName());
```
Voice channels can be mentioned with the prefix `#!`

Reply:
This method should return how the client displays something, not how to mention it. See the [docs](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/entities/Message.html#getContentDisplay())
The client doesn't use ``#!`` but instead the following: 
![Screenshot 2021-07-01 113735](https://user-images.githubusercontent.com/52353587/124102553-c0b90a80-da60-11eb-8f7e-cc363d0dbd5e.png)